src/Streamly/Coreutils/Uniq.hs

```diff
@@ -0,0 +1,126 @@
+module Streamly.Coreutils.Uniq (
+        ignCase
+      , splitOnNewLine
+      , getRepetition
+      , filterStream
+      , defaultUniqOptions
+      , UniqOptions (..)
+      , merge
+      , uniqCount
+   )
+where
+import qualified Streamly.Prelude as S
+import qualified Streamly.Internal.Data.Fold as FL
+
+import Data.Char (toLower, isSpace)
+import Control.Monad.IO.Class (MonadIO)
+import System.IO.Unsafe (unsafePerformIO)
+import Streamly.Internal.Data.Stream.StreamK.Type (IsStream)
+
+-------------------------------------------------------------------------------
+-- Record for options used with uniq
+-------------------------------------------------------------------------------
+
+{-# INLINE intMax #-}
+
+intMax :: Int
+intMax = maxBound
+
+
+data UniqOptions = UniqOptions {
+                       count :: Bool
+                     , repeated :: Bool       -- display only duplicate lines once for each group
+                     , duplicate :: Bool      -- print all duplicate lines
+                     , skipFields :: Int      -- skips first skipFields number of non-empty strings
+                     , skipChar :: Int        -- skips first skipChar number of characters (after above op)
+                     , checkChar :: Int       -- take first checkChar number of characters for comparison (after above op)
+                     , ignoreCase :: Bool     -- whether to ignore case
+                     , unique :: Bool         -- print only unique lines
+                     , delimiter :: Char      -- to separate output lines
+                  }
+
+{-# INLINE defaultUniqOptions #-}
+
+defaultUniqOptions :: UniqOptions
+defaultUniqOptions = UniqOptions True False False 0 0 intMax False True '\n'
+
+
+-------------------------------------------------------------------------------
+-- helper functions for uniq
+-------------------------------------------------------------------------------
+
+ignCase :: (IsStream t, Monad m) => Bool -> t m Char -> t m Char
```

Comment:
This changes the stream itself. We just have to ignore case for the sake of comparision.

Reply:
Oh, my bad. Thanks :+1: